django/pyrog/api/serializers/import_export.py

```diff
@@ -196,6 +196,16 @@ class Meta:
             "filters",
         ]
 
+    def validate(self, data):
+        request = self.context["request"]
+        auth_token = request.session.get("oidc_access_token")
+        fhir_api = HapiFhirAPI(auth_token)
```

Comment:
adapters must be passed when initializing the class or calling the function. We completely lose the benefit of an adapter by instantiating it like this.

Reply:
where can I instantiate this ? I need access to the current user session. I can call this by another name.

Reply:
can't you override the constructor and pass a `FhirAPI` instance to it ?
```
class MappingInputSerializer(serializers.ModelSerializer):
    def __init__(self, fhir_api: FhirAPI, *args, **kwargs):
        self.fhir_api = fhir_api
        super(MappingInputSerializer, self).__init__(*args, **kwargs)

    def validate(self, data):
         ...
         self.fhir_api.get(...)
```

Reply:
no it would force me to instantiate MappingInputSerializer myself and I'm not sure I want to that